rtc_types/src/byte_formats/rkyv_format.rs

```diff
@@ -0,0 +1,105 @@
+//! [`rkyv`] byte format helpers.
+
+use core::mem::size_of;
+
+use rkyv::{
+    archived_root,
+    ser::{
+        serializers::{BufferSerializer, BufferSerializerError},
+        Serializer,
+    },
+    Aligned, Archive, Deserialize, Infallible, Serialize, Unreachable,
+};
+
+pub fn write_array<T>(value: &T) -> Result<[u8; size_of::<T::Archived>()], BufferSerializerError>
+where
+    T: Serialize<BufferSerializer<Aligned<[u8; size_of::<T::Archived>()]>>>,
+{
+    let mut serializer = BufferSerializer::new(Aligned([0u8; size_of::<T::Archived>()]));
+    serializer.serialize_value(value)?;
+    let buf = serializer.into_inner();
+    Ok(buf.0)
```

Comment:
Another minor style nitpick. I prefer using de-structuring when possible (and when it makes sense).

This might work and be a bit more (subjectively) readable.
```
let Aligned(buf) = serializer.into_inner();
Ok(buf)
```
Not important, so feel free to mark as resolved.

Reply:
This looks much better, thanks!